### PR TITLE
Inline diagnostics and intellicode cohesion

### PR DESCRIPTION
**Pull request by akhera99**
Hola

Here is the work to handle intersecting adornments on the editor. With the added IEndOfLineAdornmentTag, the editor team raises TagsChanged for that tag when intellicode (+ other features that show up on the editor) get added. I look for those tag changed events to remove/readd my adornments.

The IEndOfLineAdornmentTag has a lot of properties that are not currently being used,  but can be used in the future if we want to have them lay together on the editor.

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs [hunk 1]
```diff
@@ -211,8 +216,9 @@ protected override void AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(Normali
 
                 var lineView = TextView.GetTextViewLineContainingBufferPosition(point.Value);
 
-                var visualElement = graphicsResult.VisualElement;
-
+                // Looking for IEndOfLineTags and seeing if they exist on the same line as where the
```
Comment (dpugh) [inline src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs:219]: This code should probably happen before creating the graphics for the inline error adornment.

You also might be able to simplify the check below to if (_endLineTagAggregator.GetTags(lineView.Extent).Any()) continue;

(the GetTags call will only return tags that intersect the span passed in.

File: src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs [hunk 2]
```diff
@@ -43,6 +43,11 @@ internal class InlineDiagnosticsAdornmentManager : AbstractAdornmentManager<Inli
             _endLineTagAggregator.BatchedTagsChanged += EndLineTagAggregator_BatchedTagsChanged;
         }
 
+        /// <summary>
+        /// Getting all tags changed events and removing all inline diagnostics to be redrawn
+        /// based on if they intersect with any IEndOfLineAdornmentTags after the layout change
+        /// has completed.
+        /// </summary>
```
Comment (dpugh) [inline src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs:50]: In the loop below, aggregating all the changedSpans into a collection of everything that changed and then calling UpdateSpans... on that might be more efficient (I'm assuming changedSpans is something similar to the editor's NormalizedSpanCollection).

File: src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs (hunk 1 above)
Comment (akhera99) [reply src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs:213]: ah, I could not remember extent. Changed, that logic is much simpler

File: src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs (hunk 2 above)
Comment (akhera99) [reply src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs:51]: now flatten the list of spans

**Review by ryzngard — APPROVED** [review]

**Comment by JoeRobich** [comment]
Might be good to run a VAL insertion to test these packages updates before merging.

**Comment by Cosifne** [comment]
Retarget this to main because 17.1 branch is not used and main is 17.1 now

**Pull request merged**